HW3/P5/label_regions.cl

```diff
@@ -80,20 +84,54 @@ propagate_labels(__global __read_write int *labels,
     old_label = buffer[buf_y * buf_w + buf_x];
 
     // CODE FOR PARTS 2 and 4 HERE (part 4 will replace part 2)
+    // grab the grandparent label
+    // if (old_label < w*h) {
+    //     buffer[buf_y * buf_w + buf_x] = labels[old_label];
+    // }
+    // if we are at the top left corner of the work group, we propogate grandparent
+    // labels across the group in a serialized fashion
+    if (lx == 0 && ly == 0) {
+        int prev, curr;
```

Comment:
Note that prev is not initialized, and so may have an arbitrary value when read later in line 101.